doc/src/build/programming-with-objects/ch5-dynamic-fields.md

```diff
@@ -0,0 +1,166 @@
+---
+title: Chapter 5 - Dynamic Fields
+---
+
+In previous chapters, we walked through various ways to use object fields to store primitive data and other objects (wrapping), but there are a few limitations to this approach:
+
+1. Object's have a finite set of fields keyed by identifiers that are fixed when its module is published.
+2. An object can become very large if it wraps several other objects. Larger objects can lead to higher gas fees in transactions. In addition, there is an upper bound on object size.
+3. As we will see in future chapters, there will be use cases where we need to store a collection of objects of heterogeneous types. Since the Move `vector` type must be templated on one single type `T`, it is not suitable for this.
```

Comment:
```suggestion
3. As we will see in future chapters, there will be use cases where we need to store a collection of objects of heterogeneous types. Since the Move `vector` type must be instantiated with one single type `T`, it is not suitable for this.
```